src/app/static/src/app/components/plots/utils.ts

```diff
@@ -162,15 +163,45 @@ const roundToPlaces = function (value: number, decPl: number) {
 // Iteratively passes through the layers of a FilterOption object to find the regional hierarchy above the supplied id
 // Takes param any for obj and returns any because it will iterate through both objects (the NestedFilterOption) and arrays (the array of child options), treating array indices as keys
 export const findPath = function (id: string, obj: any): any {
-    for (const key in obj) {
-        if (obj.hasOwnProperty(key)) {
-            if (id === obj[key]) return "";
-            else if (obj[key] && typeof obj[key] === "object") {
-                const path = findPath(id, obj[key]);
-                if (path != undefined) {
-                    return ((obj.label ? obj.label + "/" : "") + path).replace(/\/$/, '');
-                }
-            }
-        }
+  for(const key in obj) {                                         
+      if(obj.hasOwnProperty(key)) {                         
+          if(id === obj[key]) return "";                      
+          else if(obj[key] && typeof obj[key] === "object") {   
+            const path = findPath(id, obj[key]);               
+              if (path != undefined) {
+                return ((obj.label ? obj.label + "/": "") + path).replace(/\/$/, '');   
+              }              
+          }
+      }
+  }
+};
+
+export const formatOutput = function (value: number | string, format: string, scale: number, accuracy: number | null) {
+    let ans: number
+
+    if (typeof(value) === 'string'){
+        ans = parseFloat(value)
+    } else ans = value
+
+    if (!format.includes('%') && scale){
+        ans = ans * scale
+    }
+
+    if (!format.includes('%') && accuracy){
+        ans = Math.round(ans / accuracy) * accuracy
+    }
+    return numeral(ans).format(format)
+};
+
+interface Indicator {
+    indicator: string, format: string, scale: number, accuracy: number | null
+}
+
+export const findMetaData = function(indicators: Indicator[], indicatorId: string){
+    return indicators.reduce((acc: Indicator, value: Indicator) => {
```

Comment:
This is a rather unconventional usage of `reduce` which is generally used when you want to aggregate all the elements of an array in some way; I think what you want here is `find`:
```
return indicators.find(i => i.indicator == indicator.id)
```

These two calls will be slightly different - the code here using `reduce` will return the last element of the array that matches the indicator id; the code using `find` will return the first. I presume we expect there to be only one? In which case `find` would be the clearer and more efficient way. 

Just to also give a bit of context about `reduce` - this code as is would be error prone in 2 ways.
1. it would return the first element of the array if nothing matched the indicator id (since it initialises `acc` to  be the first element of the array) 
1. it would error if the array was empty (because it can't initialise `acc`)

Both of these problems would be fixed by passing `null` as the second argument to `reduce`, which would initialise `acc` as `null`.